Added value attribute to TensorVariable subclasses

fonnesbeck opened the pull request:
I hate having to remember `tag.test_value` all the time. Now all variables should return the test value using a `value` property.

Commit:
Added value attribute to TensorVariable subclasses

Commit:
Added value attribute to TensorVariable subclasses

Commit:
Added value attribute to TensorVariable subclasses

Commit:
Added value attribute to TensorVariable subclasses

Commit:
Added value attribute to TensorVariable subclasses

Commit:
Added value attribute to TensorVariable subclasses

twiecki commented:
That's nice syntactic sugar. However, I'm concerned that this might be confusing with the way pymc2 used `.value` -- as it's currently sampled value. With this change, `.value` would never change. Perhaps renaming this to `.init_value` would be more clear?

Commit:
Renamed value to init_value for TensorVariable subclasses

Commit:
Renamed value to init_value for TensorVariable subclasses

Commit:
Renamed value to init_value for TensorVariable subclasses

Commit:
Renamed value to init_value for TensorVariable subclasses

Commit:
Renamed value to init_value for TensorVariable subclasses

Commit:
Renamed value to init_value for TensorVariable subclasses

fonnesbeck commented:
Good point. I have changed that.

TravisCI seems to be failing on `conda create` for Py2.7. Not sure why.

Commit:
Debugging travis

Commit:
Debugging travis

fonnesbeck commented:
Bizarre conda failure. Only happens for 2.7, and the PATH seems correct. Any ideas?

fonnesbeck commented:
Actually, my implementation is probably not good enough because whenever a function is applied to the variable it will blow away the `init_value` property. I probably need to subclass `TensorVariable` instead.

stevenmanton commented:
@fonnesbeck I'm getting the same Travis build error for 2.7. I'll look into it, but can you let me know if you figure out why it's failing?

stevenmanton commented:
@fonnesbeck I think I [found the error](https://github.com/stevenmanton/pymc3/commit/aaae325d2b0b5a473b31220fde88c323d3bb222b). Have a look.

fonnesbeck commented:
Aha, legacy Python!

twiecki commented:
Rebased off of current master.

agoldin committed:
LKJ distribution: check if matrix is positive definite. Without this
check MCMC strays into neverland with negative eigenvalues for
covariance matrix. This is not good at all ( see
https://github.com/pymc-devs/pymc3/issues/873 ).

olafSmits committed:
Add GARCH11 model

agoldin committed:
Check if symmetric matrix is positive definite -- test.

agoldin committed:
New posdef check for WishArt

springcoil commented:
Any chance someone can rebase this and push it in again.

twiecki commented:
Done.

twiecki commented on .travis.yml:
```diff
@@ -5,6 +5,8 @@ python:
   - "3.4"
 
 before_install:
+  - sudo apt-get install liblapack-dev gfortran python-scipy libblas-dev libsuitesparse-dev swig
```
Is this still required?

twiecki commented:
We should probably also merge this as it seems to pass now. I don't think we need the travis changes though.

Commit:
Removed extraneous lines in .travis.yml

Commit:
Merge branch 'value_attr' of github.com:pymc-devs/pymc3 into value_attr

fonnesbeck commented:
Replaced by #995
